architecture/design/smart-driver.md

```diff
@@ -0,0 +1,233 @@
+<!-- Output copied to clipboard! -->
+
+
+<h1>Smart JDBC driver for YSQL</h1>
```

Comment:
Please use the markdown for Github... for example:

* If you write:
`# Smart JDBC driver`
it renders as:
# Smart JDBC driver

* For subheaders below, if you write:
* `## Motivation`
it renders as:
## Motivation

And use `### XYZ` for the third level heading.